### PR TITLE
feat: add shares account page

### PR DESCRIPTION
**Pull request by spider0061**
## Description
- Add shares module
- Add shares routing
- Add shares service
- Add share account resolver
- Add view share account component
- Add lazy loading for shares module

## Related issues and discussion
#224 

## Screenshots, if any
Screenshot from 2020-04-09 16-01-17

## Checklist
Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] If you have multiple commits please combine them into one commit by squashing them.

- [x] Read and understood the contribution guidelines at `web-app/.github/CONTRIBUTING.md`.

### CODE REVIEW

**Comment by spider0061** [comment]
> @spider0061 try lazy loading the shares module in the following way:
> Screen Shot 2020-04-09 at 10 34 25 PM
> This would make sure that breadcrumbs are rendered correctly and they convey the path traversed by the user.

@Jov03 Do I have to include lazy loading in my PR or it was just for testing because the clients module needs to be refactored first.

**Comment by spider0061** [comment]
The build error is due to timeout.

**Comment by Jov03** [comment]
> > @spider0061 try lazy loading the shares module in the following way:
> > Screen Shot 2020-04-09 at 10 34 25 PM
> > This would make sure that breadcrumbs are rendered correctly and they convey the path traversed by the user.
> 
> @Jov03 Do I have to include lazy loading in my PR or it was just for testing because the clients module needs to be refactored first.

clients modules was refactored during GSoC 2019 now group and center modules are yet to be refactored.

**Comment by spider0061** [comment]
@Jov03 Sorry I misunderstood. I tried to implement lazy loading locally but I have not included that in the PR. I'll make the required changes but I can not do that right now as I have submitted the PR link.

**Comment by Jov03** [comment]
> @Jov03 Sorry I misunderstood. I tried to implement lazy loading locally but I have not included that in the PR. I'll make the required changes but I can not do that right now as I have submitted the PR link.

No issues make the required change to your `spider0061:issue224` branch changes would be reflected automatically in your PR #723. Remember to squash commits before pushing to your branch.

**Comment by spider0061** [comment]
> > @Jov03 Sorry I misunderstood. I tried to implement lazy loading locally but I have not included that in the PR. I'll make the required changes but I can not do that right now as I have submitted the PR link.
> 
> No issues make the required change to your `spider0061:issue224` branch changes would be reflected automatically in your PR #723. Remember to squash commits before pushing to your branch.

@Jov03 Okay, I'll do that.

**Comment by spider0061** [comment]
@Jov03 added lazy loading. Please have a look.

File: src/app/shares/shares.service.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+   */
+  constructor(private http: HttpClient) { }
+
+  /**
+   * @param shareAccountId Share Account Id
+   * @returns {Observable<any>} Share Account Details
+   */
+  getShareAccount(shareAccountId: any): Observable<any> {
+    return this.http.get(`/accounts/share/${shareAccountId}`);
+  }
```
Comment (muskankhedia) [inline src/app/shares/shares.service.ts:27]: As the data is already retrieved using resolver, there is no need to make the request from the service file.

File: src/app/shares/view-share-account/view-share-account.component.ts [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+      for (const dividend of this.shareAccountData.dividends) {
+        const dividendData: any = {
+          transactionDate: dividend.postedDate,
+          amount: this.shareAccountData.currency.displaySymbol + dividend.amount,
+          transactionRef: dividend.savingsTransactionId,
+          status: dividend.status.value
+        };
+        this.dividendsTableDataSource.push(dividendData);
+      }
+    }
```
Comment (muskankhedia) [inline src/app/shares/view-share-account/view-share-account.component.ts:128]: can it be replaced with foreach()?

File: src/app/shares/view-share-account/view-share-account.component.ts [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+        if (
+          this.shareAccountData.purchasedShares[i].status.code === 'purchasedSharesStatusType.applied' &&
+          this.shareAccountData.purchasedShares[i].type.code === 'purchasedSharesType.purchased'
+        ) {
+          this.sharesPendingForApproval = true;
+          break;
+        }
+      }
+
+      for (const purchasedShare of this.shareAccountData.purchasedShares) {
```
Comment (muskankhedia) [inline src/app/shares/view-share-account/view-share-account.component.ts:76]: same

File: src/app/shares/view-share-account/view-share-account.component.ts [hunk 4]
```diff
@@ -0,0 +1,232 @@
...
+      this.choice = true;
+    }
+
+    this.chargeAction = this.shareAccountData.status.value === 'Submitted and pending approval' ? true : false;
+    this.chargePayAction = this.shareAccountData.status.value === 'Active' ? true : false;
+    this.sharesPendingForApproval = false;
+
+    /*Initialising purchased shares table data source.*/
+    if (this.shareAccountData.purchasedShares.length > 0) {
+      for (let i = 0; i < this.shareAccountData.purchasedShares.length; i++) {
```
Comment (muskankhedia) [inline src/app/shares/view-share-account/view-share-account.component.ts:66]: can it be replaced with foreach?

File: src/app/shares/view-share-account/view-share-account.component.ts [hunk 5]
```diff
@@ -0,0 +1,232 @@
...
+  dividendsTableColumns: string[] =  ['transactionDate', 'amount', 'transactionRef', 'status'];
+
+  /**
+   * Retrieves the shares account data from `resolve`.
+   * @param {SharesService} sharesService Shares Service.
+   * @param {ActivatedRoute} route Activated Route.
+   * @param {Router} router Router for navigation.
+   */
+  constructor(private route: ActivatedRoute,
+              private sharesService: SharesService,
```
Comment (muskankhedia) [inline src/app/shares/view-share-account/view-share-account.component.ts:48]: as it's not used, it can be removed.

**Pull request closed without merging**